catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/RemoteSolrCatalogProvider.java

```diff
@@ -221,133 +183,17 @@ public UpdateResponse update(UpdateRequest updateRequest) throws IngestException
 
   /** Shuts down the connection to Solr and releases resources. */
   public void shutdown() {
-    closeSolrClient();
-  }
-
-  /**
-   * Returns the current Solr server URL. The format of the URL may vary based on the current Solr
-   * configuration (embedded, external or cloud).
-   *
-   * @return current Solr URL
-   */
-  @Nullable
-  public String getUrl() {
-    return url;
-  }
-
-  /**
-   * Sets Solr's URL. The format of the URL may vary based on the current Solr configuration, e.g.,
-   * embedded, external or cloud. See the {@code system.properties} configuration file for the
-   * different options and formats. <br>
-   * Changing the URL will trigger the creation of a new Solr client using {@link #createClient()}.
-   *
-   * @param url new Solr URL
-   */
-  public void setUrl(@Nullable String url) {
-    updateClient(PropertyResolver.resolveProperties(url));
-  }
-
-  /** Forces an update of the Solr client. */
-  protected void updateClient() {
-    clientFuture = createClient();
-    client = null;
-  }
-
-  /**
-   * Request the creation of a new {@code SolrClient}.
-   *
-   * @return {@code Future} used to retrieve the new {@code SolrClient} created
-   */
-  protected abstract Future<SolrClient> createClient();
-
-  private void updateClient(@Nullable String urlValue) {
-    LOGGER.debug("New url {}", urlValue);
-
-    if (urlValue != null) {
-      if (!StringUtils.equalsIgnoreCase(urlValue.trim(), url) || getClient() == null) {
-        url = urlValue.trim();
-
-        if (getClient() != null) {
-          LOGGER.debug(
-              "Shutting down the connection manager to Solr and releasing allocated resources.");
-          closeSolrClient();
-          LOGGER.debug("Shutdown complete.");
-        }
-
-        updateClient();
-      }
-
-    } else {
-      url = null;
-    }
-  }
-
-  private void closeSolrClient() {
     LOGGER.debug("Closing connection to Solr client.");
-    if (client != null) {
-      try {
-        client.close();
-      } catch (IOException e) {
-        LOGGER.info("Unable to close Solr client", e);
-      }
-    } else if (clientFuture != null && !clientFuture.isDone() && !clientFuture.isCancelled()) {
-      clientFuture.cancel(true);
+    final SolrClient c = client;
+    try {
+      c.close();
+    } catch (IOException e) {
+      LOGGER.info("Unable to close Solr client", e);
     }
-
     LOGGER.debug("Finished closing connection to Solr client.");
```

Comment:
Not sure what we would gain? We would now have an attribute that is not final, all over the places checks for null. By keeping it this way, any calls made to it will eventually end up to the a closed client which will properly throw a not available exception. I would prefer leaving it that way and relying on the lower layer to know it is closed instead of duplicating that work here.